taxi/views.py

```diff
@@ -79,6 +82,28 @@ class CarDeleteView(LoginRequiredMixin, generic.DeleteView):
     success_url = reverse_lazy("taxi:car-list")
 
 
+class AssignDriverView(LoginRequiredMixin, View):
+    def post(self, request, pk, *args, **kwargs):
+        car = get_object_or_404(Car, pk=pk)
+        if request.user in car.drivers.all():
+            messages.info(request, "You are already a driver of this car.")
+        else:
+            car.drivers.add(request.user)
```

Comment:
In `AssignDriverView`, ensure that only `Driver` instances can be assigned to a car. You might want to check if `request.user` is an instance of `Driver` before adding them to `car.drivers`.